devito/types/sparse.py

```diff
@@ -120,6 +140,22 @@ def __subfunc_setup__(self, key, suffix, dtype=None):
         if key is not None and not isinstance(key, SubFunction):
             key = np.array(key)
 
+        # Check if already a SubFunction
+        d = self.indices[self._sparse_position]
+        if isinstance(key, SubFunction):
+            if d in key.indices:
+                # Can use as is, dimension already matches
+                if self.alias:
+                    return key._rebuild(alias=self.alias, name=name)
+                else:
+                    return key
+            else:
```

Comment:
This `else` can be removed and the contents of the block unindented by a level.

Reply:
same